File: app/schemas/quest.py
from pydantic import BaseModel

from app.core.quests import QuestType


class QuestStatusRead(BaseModel):
    id: str
    type: QuestType
    title: str
    description: str
    xp_reward: int
    # True if the *current* period (the one QuestService just evaluated --
    # "once" for one_time, "this week" for weekly/long_term) is done.
    completed: bool
    # None for one_time quests -- always set for weekly/long_term, ISO date
    # of the Monday the current period is tracked under, so the frontend
    # can show "this week" honestly rather than a vague label.
    period_start: str | None = None

File: frontend/src/types/quest.ts
export const QUEST_TYPES = ['one_time', 'weekly', 'long_term'] as const
export type QuestType = (typeof QUEST_TYPES)[number]

export const QUEST_TYPE_LABELS: Record<QuestType, string> = {
  one_time: 'Разовое',
  weekly: 'Недельное',
  long_term: 'Долгосрочное',
}

export interface QuestStatusRead {
  id: string
  type: QuestType
  title: string
  description: string
  xp_reward: number
  completed: boolean
  // Set for weekly/long_term (the Monday the current period is tracked
  // under), null for one_time -- see app/schemas/quest.py.
  period_start: string | null
}
